docs/source/migration-guides.rst

```diff
@@ -0,0 +1,258 @@
+
+.. code-block:: ts
+
+  import {expect} from 'chai';
+  import {loadFixture, deployContract} from 'ethereum-waffle';
+  import BasicTokenMock from './build/BasicTokenMock';
+
+  describe('Fixtures', () => {
+    async function fixture([wallet, other], provider) {
+      const token = await deployContract(wallet, BasicTokenMock, [
+        wallet.address, 1000
+      ]);
+      return {token, wallet, other};
+    }
+
+    it('Assigns initial balance', async () => {
+      const {token, wallet} = await loadFixture(fixture);
+      expect(await token.balanceOf(wallet.address)).to.equal(1000);
+    });
+
+    it('Transfer adds amount to destination account', async () => {
+      const {token, other} = await loadFixture(fixture);
+      await token.transfer(other.address, 7);
+      expect(await token.balanceOf(other.address)).to.equal(7);
+    });
+  });
+
+**Added automatic recognising waffle.json config without cli argument**
+
+Waffle recognises :code:`waffle.json` as default configuration file. If your configuration file is called
+:code:`waffle.json`, it's possible to use just :code:`waffle` to build contracts.
+
+*Waffle 2.5.0*
+
+.. code-block:: json
+
+  {
+    "scripts": {
+      "build": "waffle waffle.json"
+    }
+  }
+
+*Waffle 3.0.0*
+
+.. code-block:: json
+
+  {
+    "scripts": {
+      "build": "waffle"
+    }
+  }
+
+**Introduced MockProviderOptions**
+
+We added MockProviderOptions. It will be convenient in the future, when the provider may need some
+other options than :code:`ganacheOptions`.
+
+*Waffle 2.5.0*
+
+.. code-block:: ts
+
+  import {expect} from 'chai';
+  import {Wallet} from 'ethers';
+  import {MockProvider} from 'ethereum-waffle';
+  import {deployToken} from './BasicToken';
+
+  describe('INTEGRATION: MockProvider', () => {
+    it('accepts options', () => {
+      const original = Wallet.createRandom();
+      const provider = new MockProvider({
+          accounts: [{balance: '100', secretKey: original.privateKey}]
+      });
+      const wallets = provider.getWallets();
+      expect(wallets.length).to.equal(1);
+      expect(wallets[0].address).to.equal(original.address);
+    });
+  });
+
+*Waffle 3.0.0*
+
+.. code-block:: ts
+
+  import {expect} from 'chai';
+  import {Wallet} from 'ethers';
+  import {MockProvider} from 'ethereum-waffle';
+  import {deployToken} from './BasicToken';
+
+  describe('INTEGRATION: MockProvider', () => {
+    it('accepts options', () => {
+      const original = Wallet.createRandom();
+      const provider = new MockProvider({
+        ganacheOptions: {
+          accounts: [{balance: '100', secretKey: original.privateKey}]
+        }
+      });
+      const wallets = provider.getWallets();
+      expect(wallets.length).to.equal(1);
+      expect(wallets[0].address).to.equal(original.address);
+    });
+  });
+
+**Dropped support for contract.interface**
+
+We dropped support for contract.interface, because it duplicated contract ABI. Now we support just :code:`contract.abi`.
```

Comment:
Several reasons here actually. Originally waffle 1 only had interface, then we replaced `interface` with `abi`, but it was a breaking change. So we added an option to output a legacy version that included `interface`. Unfortunately due to a bug in the code both keys were always outputted. In waffle 3 we dropped interface not only because it is duplicated, but `interface` is a keyword in typescript so you cannot destructure to it.